src/tensor.rs

```diff
@@ -233,6 +253,57 @@ impl Tensor {
         Ok(Self(Arc::new(tensor_)))
     }
 
+    pub fn matmul(&self, rhs: &Self) -> Result<Self> {
+        let a_dims = self.shape().dims();
+        let b_dims = rhs.shape().dims();
+
+        let dim = a_dims.len();
+
+        // if dim < 2 {
```

Comment:
Re-enable this test as part of the `if b_dims.len() < 2`, i.e. `if dim < 2 || b_dims.len() < 2`?